src/core/moves.h

```diff
@@ -28,6 +43,40 @@ allowednextmove(uint8_t *moves, uint8_t n)
 	return axis[1] != axis[2] || base[0] != base[2];
 }
 
+static uint32_t 
+disable_moves(uint32_t current_result, uint8_t base_index)
+{
+	return current_result & ~((1 << base_index) | (1 << (base_index + 1)) | (1 << (base_index + 2)));
```

Comment:
Più corto (ma vedi tu se lo preferisci): `current_result & ~(7 << base_index)`

Per coerenza con le altre funzioni userei come modificatore `_static_inline` e aggiungerei un prototipo per la funzione in cima al file (es. riga 9).